### PR TITLE
CDAP-15816: Mainframe reader plugin validation.

### PR DESCRIPTION
**Pull request by AntonGensitskiy**
JIRA: https://issues.cask.co/browse/CDAP-15816
WIKI: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation

In scope of this PR:

- integrated with new validation API

### CODE REVIEW

File: src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+  public static final String COPYBOOK_CONTENTS = "copybookContents";
+  public static final String DROP = "drop";
+  public static final String KEEP = "keep";
+  public static final String MAX_SPLIT_SIZE = "maxSplitSize";
+  public static final String CODEPAGE = "codepage";
+  public static final String CHARSET = "charset";
+
+  public static final long DEFAULT_MAX_SPLIT_SIZE_IN_MB = 1;
+  public static final long CONVERT_TO_BYTES = 1024 * 1024;
+  public static final Pattern RTRIM = Pattern.compile("\\s+$");
```
Comment (aonischuk) [inline src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java]: Can we rename it to something more clear.

File: src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+    this.drop = drop;
+    this.keep = keep;
+    this.maxSplitSize = maxSplitSize;
+    this.codepage = codepage;
+    this.charset = charset;
+    this.replacements = replacements;
+  }
+
+  private MainframeSourceConfig(Builder builder) {
+    super("CopybookReader");
```
Comment (aonischuk) [inline src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java]: Why are we hardcoding the reference name?

File: src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    }
+    if (!Strings.isNullOrEmpty(copybookContents)) {
+      String copybook = getCopyBookContents();
+      int lineNo = 0;
+      for (String line : copybook.split("\\r?\\n")) {
+        ++lineNo;
+        for (int i = line.length() - 1; i > 0; i--) {
+          String trimmed = RTRIM.matcher(line).replaceAll("");
+          if (trimmed.length() > 72) {
+            String message = String.format("Copybook line %d exceeds the maximum length of 72 characters. " +
```
Comment (aonischuk) [inline src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java]: '%d'

File: src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java (hunk 1 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java]: Done.

File: src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java (hunk 2 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java]: It was based implementation. I didn't found reason why should be used hardcoded value.
Updated for using 'referenceName' property.

File: src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java (hunk 3 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/batch/mainframe/reader/MainframeSourceConfig.java]: Done.

**Pull request closed without merging**